sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -559,6 +562,248 @@ def set_read_only(
+            raise e(message=error.message, response=error.response)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def get_snapshot(
+        self,
+        name: str,
+        *,
+        fields: Optional[List[str]] = None,
+        **kwargs
+    ) -> Snapshot:
+        """Get a configuration setting snapshot.
+
+        :param name: The name of the configuration setting snapshot to retrieve.
+        :type name: str
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :return: The Snapshot returned from the service.
+        :rtype: :class:`~azure.appconfiguration.Snapshot`
+        :raises: :class:`HttpResponseError`, :class:`ClientAuthenticationError`
+        """
+        error_map = {401: ClientAuthenticationError}
+        try:
+            return self._impl.get_snapshot(
+                name=name,
+                if_match=None,
+                if_none_match=None,
+                select=fields,
+                **kwargs
+            )
+        except HttpResponseError as error:
+            e = error_map[error.status_code]
+            raise e(message=error.message, response=error.response)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshots(
+        self,
+        *,
+        name: Optional[str] = None,
+        fields: Optional[List[str]] = None,
+        status: Optional[List[str]] = None,
+        **kwargs
+    ) -> ItemPaged[Snapshot]:
+        """List the configuration setting snapshots stored in the configuration service, optionally filtered by
+        snapshot name and status.
+
+        :keyword str name: Filter results based on snapshot name.
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :keyword list[str] status: Filter results based on snapshot keys.
+        :return: An iterator of :class:`~azure.appconfiguration.Snapshot`
+        :rtype: ~azure.core.paging.ItemPaged[~azure.appconfiguration.Snapshot]
+        :raises: :class:`HttpResponseError`, :class:`ClientAuthenticationError`
+        """
+        error_map = {401: ClientAuthenticationError}
+        try:
+            return self._impl.get_snapshots(  # type: ignore
+                name=name,
+                select=fields,
+                status=status,
+                **kwargs
+            )
+        except HttpResponseError as error:
+            e = error_map[error.status_code]
+            raise e(message=error.message, response=error.response)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshot_configuration_settings(self, name: str, **kwargs) -> ItemPaged[ConfigurationSetting]:
```

Comment:
While it's not blocking to this PR, I still think I would advocate we expose this API as an overload to the existing `list_configuration_settings` method rather than a separate method.